application-licensing-licensor/application-licensing-licensor-api/pom.xml

```diff
@@ -38,6 +38,8 @@
     <xwiki.extension.namespaces>{root}</xwiki.extension.namespaces>
     <checkstyle.suppressions.location>${basedir}/src/main/checkstyle/checkstyle-suppressions.xml</checkstyle.suppressions.location>
     <xwiki.jacoco.instructionRatio>0.50</xwiki.jacoco.instructionRatio>
+    <!-- Added for the licensor-ui dependency, since a JAR extension shouldn't depend on a XAR one. -->
+    <xwiki.enforcer.banneddependencytype-xar.skip>true</xwiki.enforcer.banneddependencytype-xar.skip>
```

Comment:
Why does it depend on the UI module?

Reply:
From what I understand, this is how the `licensor` was designed, so each pro app has the `api` module as dependency, which has the `ui`. This, as we know, makes the pro (licensed) apps to be identified by the `api` dependency and I guess that they didn't used the `ui` module instead since you might want to depend on it for other usecases. Or simply it wasn't a best practice at that moment to not have apis depending on xar and the current solution felt more natural.
Anyway, I don't think it's an easy fix to avoid it now, maybe only adding also the `ui` dependency besides the `api` one, but I'm not very fond of this. WDYT?

Reply:
So, If i understand correctly, the pro apps have a dependency to the `licensor-api` which marks them as an app that requires a license. The `licensor-api` has a dependency to the ui module in order to bring all the xwiki pages when installing the the mentioned pro app.

In order to get rid of this dependency to the ui module, we would need to update the code that checks whether an app depends on the `licensor-api` and update all the pro apps to depend on the `licensor-ui`?

If so, it does sound like plenty of work that might cause some complications. But it might also be fairly simple to do. (unless im missing some implications).

I'm not a fan of also adding the `ui` dependency. Doing that or skipping the enforcer is kind of the same thing - a hack that needs a comment as an explanation.

I think its ok to leave this as is, for now. But it would be useful to have an issue for this.